NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Jg create tag 14

# Description
[Ticket number 14](https://trello.com/c/sarK3PGJ/14-14-create-a-tag) added the  functionality for adding a new tag.

#Updates included. 
-client->src->components-> `AddTagForm was created
-client->src->components-> `AddTagForm` was added as a child to TagList.js 
-client->src->Managers->TagManager was updated with `AddTag` function (more functions to be added later).


# Testing Instructions

1.  `git add` and `git commit` whatever you're working on on your branch
2.  `git checkout main`
3. ` git fetch --a`
4.  `git checkout jg-ViewAllTags-13 ` to switch to my branch.
5. If you don't see the most recent version of the files, try` git pull origin jg-ViewAllTags-13`.
6. If you haven't created the Tabloid FullStack database, please do so.
7.  Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
8.  Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9. Execute Tabloid from inside of Visual Studio by clicking on the solid green arrow. This will launch Swagger and run the back end of the application to provide the API that will be utilized to view the front end. 
10. From your terminal run npm start from the `client` directory within `tabloidfullstack-cranberry-sauce`
11.  Step 10 will bring up the front end of the application in a browser window titled React App. In the url of that browser page, please click on TagManagement from the Nav bar.
12.  You should see a list of Tag names printed on the that include the following from the seed data: 
-C#
-Funny
-JavaScript
-Scary
13. Above the list of tags is a form to add a tag. Please type a word into that form and click Save Tag button. You should see the new tag appear in the list in its alphabetized spot. If so, the feature functions.   

Please note: I chose to keep the form to create a tag on the same page as the tag list rather than navigating the user to another page because it is just one single field. This is a slight divergence from the ticket. Please let me know of you prefer it this way or feel it should be on a different page. 

# Checklist:

- [] I have commented my code, particularly in hard-to-understand areas
